### PR TITLE
Use the same server to launch emacs-everywhere

### PR DESCRIPTION
**Pull request by theia-jane**
The current implementation assumes that emacs-everywhere should be using the default emacs daemon running. I run multiple instances of emacs daemon for various purposes and ran into issues with this.

You can tell what type of server is running from variables and from that determine the right command line flag to use when launching `emacsclient`. I pulled this logic mostly from the `with-editor` package, but did consolidate the TCP server type to use the flag instead of as an environment variable.

The bottom line of what this PR enables is:

    emacs --daemon=everywhere
    emacsclient -s everywhere -e '(emacs-everywhere)'

Where the final emacs-everywhere frame that opens is run in the `everywhere` emacs daemon and not the default daemon.

Let me know if you have any questions or want changes! Thanks!

### CODE REVIEW

**Comment by tecosaur** [comment]
This looks good to me, thanks.
